components/clientComponents/forms/TextInput/TextInputPlugin.tsx

```diff
@@ -0,0 +1,84 @@
+// Update the path below to the correct location of ComponentPlugin
+import { ComponentPlugin } from "@lib/components/ComponentPlugin";
+import { TextInput } from "./TextInput";
+import { FormElementTypes } from "@lib/types";
+import { getLocalizedProperty } from "@lib/utils";
+import { Label, Description } from "@clientComponents/forms";
+
+export const TextInputPlugin: ComponentPlugin = {
+  meta: {
+    type: FormElementTypes.textField,
+  },
+  render({ element, lang }) {
+    const id = element.subId ?? element.id;
+    const isRequired: boolean = element.properties.validation
+      ? element.properties.validation.required
+      : false;
+
+    const descriptionPerLocale = element.properties[getLocalizedProperty("description", lang)];
```

Comment:
Maybe we should have helpers for pulling descriptions titles etc...  ?

Just thinking about how repetitive this is going to get.

Reply:
Same with the Label, I've just been thinking through how I want to do it in case we need to break a label out of an element in other ways.

Reply:
I went with just making a linked component, but it feels a bit ugly. I'll keep thinking, and if anyone has a better idea.